speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/backend/SQLDatabaseStorage.cpp

```diff
@@ -0,0 +1,160 @@
+#include <iostream>
+#include "SQLDatabaseStorage.h"
+#include "sqlite3.h"
+#include <string>
+#include <sstream>
+#include "../rppUtils/RppUtils.hpp"
+
+#define DATABASE_THROW(db, error)                     \
+    if (rc)                                           \
+    {                                                 \
+        std::string errorString(error);               \
+        sqlite3_close(db);                            \
+        remove(databasePath.c_str());\
+        if (zErrMsg != nullptr)                       \
+        {                                             \
+            std::string s(zErrMsg);                   \
+            errorString +=                            \
+                "\n\tReturned with error: "           \
+                + s;                                  \
+        }                                             \
+        errorString += "\n";                          \
+        sqlite3_free(zErrMsg);                        \
+        input.close();                                \
+        throw std::runtime_error(errorString.c_str());\
+    };
+
+SQLDatabaseStorage::SQLDatabaseStorage()
+{
+    // player info data
+    m_headerTypes["SteerCmd"] = "SteerCmd float NOT NULL";
+    m_headerTypes["AccelCmd"] = "AccelCmd float NOT NULL";
+    m_headerTypes["BrakeCmd"] = "BrakeCmd float NOT NULL";
+    m_headerTypes["ClutchCmd"] = "ClutchCmd float NOT NULL";
+
+    // environment data
+    m_headerTypes["TimeOfDay"] = "TimeOfDay int NOT NULL";
+    m_headerTypes["Clouds"] = "Clouds int NOT NULL";
+    m_headerTypes["Rain"] = "Rain int NOT NULL";
+
+    // car data
+    m_headerTypes["Speed"] = "Speed float NOT NULL";
+    m_headerTypes["TopSpeed"] = "TopSpeed float NOT NULL";
+    m_headerTypes["Gear"] = "Gear int NOT NULL";
+    m_headerTypes["Headlights"] = "Headlights bool NOT NULL";
+
+    // track position data
+    m_headerTypes["Offroad"] = "Offroad bool NOT NULL";
+    m_headerTypes["ToStart"] = "ToStart float NOT NULL";
+    m_headerTypes["ToRight"] = "ToRight float NOT NULL";
+    m_headerTypes["ToMiddle"] = "ToMiddle float NOT NULL";
+    m_headerTypes["ToLeft"] = "ToLeft float NOT NULL";
+
+    // AI decision data
+    m_headerTypes["SteerDecision"] = "SteerDecision float NOT NULL";
+    m_headerTypes["BrakeDecision"] = "BrakeDecision float NOT NULL";
+}
+
+// Create a callback function
+int callback(void *NotUsed, int argc, char **argv, char **azColName){
+
+
+    // Return successful
+    return 0;
+}
+
+void SQLDatabaseStorage::StoreData(const std::string p_filePath)
```

Comment:
StoreData seems like a long function, I would add private functions and c functions to make this smaller and more readable for other people.